core/extension/gdextension.cpp

```diff
@@ -787,9 +787,14 @@ Error GDExtension::open_library(const String &p_path, const String &p_entry_symb
 	}
 
 	GDExtensionInitializationFunction initialization_function = (GDExtensionInitializationFunction)entry_funcptr;
-	GDExtensionBool ret = initialization_function(&gdextension_get_proc_address, this, &initialization);
+	return initialize_extension_function(initialization_function, p_entry_symbol);
+}
 
-	if (ret) {
+Error GDExtension::initialize_extension_function(GDExtensionInitializationFunction initialization_function, const String &p_entry_symbol) {
+	if (library == nullptr) {
+		embedded = true;
+	}
+	if (initialization_function(&gdextension_get_proc_address, this, &initialization)) {
 		level_initialized = -1;
 		return OK;
 	} else {
```

Comment:
GitHub won't let me comment on it directly, because it's not in the diff, but in this `else` block there is this:

```
		ERR_PRINT("GDExtension initialization function '" + p_entry_symbol + "' returned an error.");
		OS::get_singleton()->close_dynamic_library(library);
		return FAILED;
```

We probably don't want to call `OS::get_singleton()->close_dynamic_library(library);` here if `library` is null.

Reply:
Good catch, will take a look.

Reply:
Moved to GDExtensionLoader implementation.